Desktop/analysis/analyses.h

```diff
@@ -187,7 +187,6 @@ private slots:
 	std::map<size_t, Analysis*>		_analysisMap;
```

Comment:
This looks like a great candidate for `std::map<size_t, std::unique_ptr<Analysis>>`, or `std::shared_ptr<>` if you really need it. Then, you can just deconstruct them, and don't worry about the `delete` operation, and everything is suddenly great! I mean, of course not, but I feel this is a very good place to start thinking about it. You need a bit of refactoring to make sure that deconstructors are working properly, but then they are safer and more pleasant to work with! 🌻

Reply:
Well, I like having some control over when things are constructed and deconstructed.
Especially as there are only a few classes where we do this manually anyway. Most of the time we use the whole qobject tree to clean up.